src/RedeemFei.sol

```diff
@@ -0,0 +1,160 @@
+     *   is a different wallet
+     * @param referralCode Code used to register the integrator originating the operation, for potential rewards.
+     *   0 if the action is executed directly by the user, without any middle-man
+     **/
+    function deposit(
+        address asset,
+        uint256 amount,
+        address onBehalfOf,
+        uint16 referralCode
+    ) external;
+}
+
+interface IEcosystemReserveController {
+    /**
+     * @notice Proxy function for ERC20's approve(), pointing to a specific collector contract
+     * @param collector The collector contract with funds (Aave ecosystem reserve)
+     * @param token The asset address
+     * @param recipient Allowance's recipient
+     * @param amount Allowance to approve
+     **/
+    function approve(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+
+    /**
+     * @notice Proxy function for ERC20's transfer(), pointing to a specific collector contract
+     * @param collector The collector contract with funds (Aave ecosystem reserve)
+     * @param token The asset address
+     * @param recipient Transfer's recipient
+     * @param amount Amount to transfer
+     **/
+    function transfer(
+        address collector,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external;
+}
+
+event PsmRedeem();
+event notPsmRedeem();
+
+contract RedeemFei is IProposalGenericExecutor {
+
+    address public constant FEI = 0x956F47F50A910163D8BF957Cf5846D573E7f87CA;
+
+    address public constant A_FEI = 0x683923dB55Fead99A79Fa01A27EeC3cB19679cC3;
+
+    address public constant AAVE_MAINNET_RESERVE_FACTOR = 0x464C71f6c2F760DdA6093dCB91C24c39e5d6e18c;
+
+    IFixedPricePSM public constant DAI_FIXED_PRICE_PSM = IFixedPricePSM(0x2A188F9EB761F70ECEa083bA6c2A40145078dfc2);
+
+    IEcosystemReserveController public constant AAVE_ECOSYSTEM_RESERVE_CONTROLLER =
+        IEcosystemReserveController(0x3d569673dAa0575c936c7c67c4E6AedA69CC630C);
+
+    ILendingPool public constant AAVE_LENDING_POOL = ILendingPool(0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9);
+
+    function execute() external override {
+        uint256 aFeiBalance = 300_000 ether;
+
+        AAVE_ECOSYSTEM_RESERVE_CONTROLLER.transfer(
+            AAVE_MAINNET_RESERVE_FACTOR,
+            A_FEI,
+            address(this),
+            aFeiBalance
+        );
+
+        AAVE_LENDING_POOL.withdraw(FEI, aFeiBalance, address(this));
+        
+        uint256 feiBalance = IERC20(FEI).balanceOf(address(this));
+
+        // The minimum amount of DAI we are willing to receive after redeeming all our FEI.
+        // PSM hardcodes 1 DAI = 1 FEI & takes a 3 bps redeem fee
+        // so we subtract a 3bps fee from our FEI balance
+        // https://etherscan.io/address/0x2A188F9EB761F70ECEa083bA6c2A40145078dfc2#readContract function 31. redeemFeeBasisPoints 
+        uint256 minBalance = feiBalance - (feiBalance / 1000 * 3);
```

Comment:
Two comments here:
1. Since it's basis points, it should be `10_000` and not `1000`. 
2. Always multiply first before dividing to prevent loss of precision. 

So final formula should be:
`uint256 minBalance = feiBalance - (feiBalance * 3 / 10000);`

Reply:
Unless I'm misunderstanding something related to point 1.

Reply:
Also is the 3 basis points something that can change? If so, then I would recommend dynamically pulling that value from the contract rather than hardcoding as 3 here.